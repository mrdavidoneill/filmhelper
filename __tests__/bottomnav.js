import React from "react";
import { render } from "@testing-library/react";
import BottomNavBar from "@/components/bottomnav";

const mockRouter = {
  pathname: "/discover",
};

jest.mock("next/router", () => ({
  useRouter: jest.fn().mockImplementation(() => mockRouter),
}));

describe("BottomNavBar component", () => {
  it("renders correctly for /discover path", () => {
    mockRouter.pathname = "/discover";
    const { asFragment } = render(<BottomNavBar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly for /watchlist path", () => {
    mockRouter.pathname = "/watchlist";
    const { asFragment } = render(<BottomNavBar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly for /ratings path", () => {
    mockRouter.pathname = "/ratings";
    const { asFragment } = render(<BottomNavBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
