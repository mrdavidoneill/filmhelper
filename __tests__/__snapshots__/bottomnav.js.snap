// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNavBar component renders correctly for /discover path 1`] = `
<DocumentFragment>
  <div
    class="animate-rise MuiBox-root css-0"
  >
    <div
      class="MuiBottomNavigation-root w-full css-gumrph-MuiBottomNavigation-root"
    >
      <a
        class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
        href="discover"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MovieIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m18 4 2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label Mui-selected css-imwso6-MuiBottomNavigationAction-label"
        >
          Discover
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
        href="watchlist"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VisibilityIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          Watch List
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
        href="ratings"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RateReviewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 14v-2.47l6.88-6.88c.2-.2.51-.2.71 0l1.77 1.77c.2.2.2.51 0 .71L8.47 14H6zm12 0h-7.5l2-2H18v2z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          My Ratings
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BottomNavBar component renders correctly for /ratings path 1`] = `
<DocumentFragment>
  <div
    class="animate-rise MuiBox-root css-0"
  >
    <div
      class="MuiBottomNavigation-root w-full css-gumrph-MuiBottomNavigation-root"
    >
      <a
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
        href="discover"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MovieIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m18 4 2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          Discover
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
        href="watchlist"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VisibilityIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          Watch List
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
        href="ratings"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RateReviewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 14v-2.47l6.88-6.88c.2-.2.51-.2.71 0l1.77 1.77c.2.2.2.51 0 .71L8.47 14H6zm12 0h-7.5l2-2H18v2z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label Mui-selected css-imwso6-MuiBottomNavigationAction-label"
        >
          My Ratings
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`BottomNavBar component renders correctly for /watchlist path 1`] = `
<DocumentFragment>
  <div
    class="animate-rise MuiBox-root css-0"
  >
    <div
      class="MuiBottomNavigation-root w-full css-gumrph-MuiBottomNavigation-root"
    >
      <a
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
        href="discover"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MovieIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m18 4 2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          Discover
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
        href="watchlist"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VisibilityIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label Mui-selected css-imwso6-MuiBottomNavigationAction-label"
        >
          Watch List
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <a
        class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
        href="ratings"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="RateReviewIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 14v-2.47l6.88-6.88c.2-.2.51-.2.71 0l1.77 1.77c.2.2.2.51 0 .71L8.47 14H6zm12 0h-7.5l2-2H18v2z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
        >
          My Ratings
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
